### DISCUSSION TITLE
Include a html file in quarto website

### DISCUSSION THREAD

**Discussion by jwang-lilly** [quarto-dev/quarto-cli#2758] [Q&A]
Hi, 

  I have an HTML file produced from a .qmd file using a number of R packages.  I also have a quarto website, and I'd like to include this HTML file as it is, not to 1) render the .qmd file within the website, using the freeze feature because I don't wish to "pollute" the website with all those R packages; 2) include the HTML as a block because the file is fairly large.  Hopefully, I can include this HTML file name in a good location such as .yml and have the content shown by clicking a tab in the nav bar.

Any advice is greatly appreciated.

**Comment by eitsupi** [comment]
You can use `resources` to including files.
https://quarto.org/docs/websites/website-tools.html#site-resources

**Reply by jwang-lilly** [reply]
Thank you to @eitsupi for the suggestion.  The caveat of this approach is the linked HTML file becomes entirely independent of the website, not embedded within the website.  It is like providing a link to rstudio.com home page, having nothing to do with my website.  This is not my intention.  I hope to find a way to have the HTML displayed within the framework of the website, with the usual navigational support.

**Reply by cscheid** [reply]
That use case is not supported by quarto, because we cannot know the structure of your file. If you want to have navigational support, then quarto needs to have access to the .qmd file, from which we can extract the metadata.

**Reply by jwang-lilly** [reply]
thanks much @cscheid

**Comment by jwang-lilly** [comment]
@dragonstyle, do you have any advice on this question?

I have a quarto website utilizing hybrid navigation following the quarto.org/docs/websites/website-navigation.html example. I have a self-contained large HTML file and I would like to include it as is within this website.  I hope to display it by clicking on a sidebar item.  Using the example quarto.org/docs/websites/website-navigation.html on Hybrid Navigation section, ProjectX website -> Tutorials on Navbar -> Tutorial 1 on sidebar => my HTML file display.  I'd like the top navbar  (possibly the sidebar as well) to be in place as usual and my HTML to be displayed in the content area.  Any other suggestions are welcome. Thanks much!

**Reply by dragonstyle** [reply]
I can think of one thing you could try -make a `qmd` file and include an iframe on the page pointing to the document (and using js to size it appropriately). This requires the `qmd` file to provide the metadata required for Quarto and provide the `framing` elements around the document. 

Since the self contained document already contains a complete HTML document, I don't know of a way to inline that without something like a pre-render script that would parse the HTML, extract the body (or some target interior element) and place that in a `qmd` file.

**Reply by jwang-lilly** [reply]
thanks much @dragonstyle

**Reply by lnalborczyk** [reply]
Hi, have you found (and implemented) a solution to this issue? Is a code example available somewhere online? I would also like to redirect to a HTML file (a bookdown gitbook) located in a subfolder of my quarto website...

**Reply by cameronrutherford** [reply]
Same here - would be quite convenient as png snippets of websites can be buggy, and saving an html copy renders better.

**Reply by cameronrutherford** [reply]
Just answered my own question!

A simple way of doing this is:

```md
<iframe src="./path/to/custom.html"></iframe>
```
Just ensure the html is pathed to correctly, considering RPATHs and whatnot

**Comment by jmhimara** [comment]
Idk if this would solve the problem, but you could try to the HTML code of the other page into an `{=html}` code block.

**Comment by stefano-tronci** [comment]
Hi!

I bumped into this while searching for a solution for a similar problem. I specifically wanted to embed HTML from file (a [drawio](https://www.drawio.com/) diagram exported to HTML). I managed to put together a solution I am happy with, so I thought I would share it.

Use a `python` code block like this:

````python

```{python}
from IPython.display import IFrame

IFrame(src='/path/to/file.html', width='100%', height=600)
```

````

You might need to adjust the height to make it look nice.  I pretty much just copied [this](https://stackoverflow.com/a/52338192). Generally speaking, I tend to find that what works in Jupyter will, perhaps unsurprisingly, work in Quarto as well.

**Reply by mcanouil** [reply]
You don't need to involve computation to make an iframe.

```markdown
![](path/to/file.html){width="100%" height="600px"}
```
